Lib/http/client.py

```diff
@@ -137,6 +137,15 @@
 _is_legal_header_name = re.compile(rb'[^:\s][^:\r\n]*').fullmatch
 _is_illegal_header_value = re.compile(rb'\n(?![ \t])|\r(?![ \t\n])').search
 
+# These characters are not allowed within http URL paths.
+#  https://tools.ietf.org/html/rfc3986#section-3.3
```

Comment:
Would you mind to describe a little bit what are these characters? It looks like control characters (0..31), space (32) and another control character (127).

Reply:
space is not validated. There is `test.test_urllib.urlopen_HttpTests.test_userpass_inurl_w_spaces` that fails if space is validated. golang also didn't check for space.

Reply:
The golang code does not consider space to be a control char (via `< ' '` rather than `<=`). See https://github.com/golang/go/commit/f1d662f34788f4a5f087581d0951cdf4e0f6e708) in their implementation for the CVE.

But space is clearly a forbidden character in a url (% or + encoding required) as the GET line is space delimited.  So I'm keeping out exclusion of space in place.

In our test_userpass_inurl_w_spaces the user@pass has been parsed and removed from the url and turned into an authorization header long before it gets into the http.client code that does the checking so this does not impact usernames or passwords with spaces.